metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_schema.py

```diff
@@ -86,6 +106,64 @@ class SnowflakeDatabase:
     comment: Optional[str]
     last_altered: Optional[datetime] = None
     schemas: List[SnowflakeSchema] = field(default_factory=list)
+    tags: Optional[List[SnowflakeTag]] = None
+
+
+class _SnowflakeTagCache:
+    def __init__(self) -> None:
+        self._database_tags: Dict[str, List[SnowflakeTag]] = defaultdict(list)
+        self._schema_tags: Dict[str, Dict[str, List[SnowflakeTag]]] = defaultdict(
+            lambda: defaultdict(list)
+        )
+        self._table_tags: Dict[
+            str, Dict[str, Dict[str, List[SnowflakeTag]]]
+        ] = defaultdict(lambda: defaultdict(lambda: defaultdict(list)))
+        self._column_tags: Dict[
+            str, Dict[str, Dict[str, Dict[str, List[SnowflakeTag]]]]
+        ] = defaultdict(
+            lambda: defaultdict(lambda: defaultdict(lambda: defaultdict(list)))
+        )
+
+    def add_database_tag(self, db_name: str, tag: SnowflakeTag) -> None:
+        self._database_tags[db_name].append(tag)
+
+    def get_database_tags(self, db_name: str) -> List[SnowflakeTag]:
+        return self._database_tags.get(db_name, [])
```

Comment:
since we're using a defaultdict, we don't need `.get()` and can just access the field directly right?